src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests_IEventReferenceOperation.vb

```diff
@@ -0,0 +1,71 @@
+﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+Imports Microsoft.CodeAnalysis.Test.Utilities
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.UnitTests.Semantics
+
+    Partial Public Class IOperationTests
+        Inherits SemanticModelTestBase
+
+        <CompilerTrait(CompilerFeature.IOperation)>
+        <Fact()>
+        Public Sub IEventReference_AddHandlerSharedEvent()
+            Dim source = <![CDATA[
+Option Strict On
+Imports System
+
+Module M1
+    Class C1
+        Shared Event E1 As EventHandler
+        Shared Sub S2()
+            AddHandler E1, Sub(sender, args)'BIND:"E1"
+                           End Sub
+        End Sub
+    End Class
+End Module]]>.Value
+
+            Dim expectedOperationTree = <![CDATA[
+IEventReferenceOperation: Event M1.C1.E1 As System.EventHandler (Static) (OperationKind.EventReference, Type: System.EventHandler) (Syntax: 'E1')
+  Instance Receiver: 
+    null
+]]>.Value
+
+            Dim expectedDiagnostics = String.Empty
+
+            VerifyOperationTreeAndDiagnosticsForTest(Of IdentifierNameSyntax)(source, expectedOperationTree, expectedDiagnostics)
+        End Sub
+
+        <CompilerTrait(CompilerFeature.IOperation)>
+        <Fact()>
+        Public Sub IEventReference_AddHandlerSharedEventWithInstanceReference()
```

Comment:
>IEventReference_AddHandlerSharedEventWithInstanceReference [](start = 19, length = 58)

Please make sure we have tests for accessing an event (static and instance) of off a type

Reply:
Done.

---
In reply to: [149249320](https://github.com/dotnet/roslyn/pull/22972#discussion_r149249320) [](ancestors = 149249320)